main.go

```diff
@@ -124,8 +138,18 @@ func mainerr() error {
 		fmt.Sprintf("--volume=%s:/init", binary),
 		"--entrypoint=/init",
 
-		// User uid and git for GOPATH and GOCACHE volume mappings
+		// User uid and git for GOPATH and GOCACHE volume mappings.
 		fmt.Sprintf("--user=%v:%v", syscall.Getuid(), syscall.Getgid()),
+
+		// Also mount host files so the container can know what user that is.
+		"--volume=/etc/group:/etc/group:ro",
```

Comment:
These make me nervous. Why are they required?

Reply:
If the guest sees UID 1000 and GID 1000 but its passwd and group files do not have entries for such IDs, then all programs fall back to "unknown user", which is terribly broken. If you simply fire up bash, you'll see the problem:

```
$ docker run -it --user $UID:$GID debian:11.1-slim bash
I have no name!@58274c25f365:/$ whoami
whoami: cannot find name for user ID 1000
```

whereas:

```
$ docker run -it --user $UID:$GID --volume=/etc/group:/etc/group:ro --volume=/etc/passwd:/etc/passwd:ro --volume=/etc/shadow:/etc/shadow:ro debian:11.1-slim bash
mvdan@ece2efbc11bd:/$ whoami
mvdan
```

Note that passwd and group simply list the users and groups, and don't contain any particularly sensitive information. It's only `shadow` that contains encrypted/hashed passwords. Perhaps we can omit shadow for now, to be conservative against potentially malicious guest images? At least for my needs, I just need to be able to query my own user and its permissions, not log into the account or otherwise use the password.

Reply:
IMO, the alternative to all this if you want to be super-conservative is to not try to map your host's UID/GID into your guest. But then that would mean disabling your feature entirely :)

Reply:
> Perhaps we can omit shadow for now, to be conservative against potentially malicious guest images? At least for my needs, I just need to be able to query my own user and its permissions, not log into the account or otherwise use the password.

Done, and updated the code docs.

Reply:
LGTM!